### PR TITLE
[TIMOB-24266] Expose application-delegates to Hyperloop / native modules

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24266

### CODE REVIEW

**Review by vijaysingh-axway — COMMENTED** [review]
It will be good if we mention this capability in document also.

**Comment by hansemannn** [comment]
Good point, will draft a document for it today!

**Comment by hansemannn** [comment]
@vijaysingh-axway The guide is now drafted (link later). Can you do the review already?

File: iphone/Classes/NetworkModule.h [hunk 1]
```diff
@@ -23,7 +23,7 @@ typedef enum {
   TLS_VERSION_1_2
 } TLSVersion;
 
-@interface NetworkModule : TiModule {
+@interface NetworkModule : TiModule <UIApplicationDelegate> {
```
Comment (vijaysingh-axway) [inline iphone/Classes/NetworkModule.h]: Every thing looks good and working fine. One question -
If we have to use the UIApplicationDelegate methods in module, we have to include UIApplicationDelegate in module interface like below -

`@interface TestAppceleratorModule : TiModule<UIApplicationDelegate>
@end`

Isn't it better to include UIApplicationDelegate in TiModule ?

Comment (hansemannn) [reply iphone/Classes/NetworkModule.h]: The main reason was that I tried to not set the delegate if it's not actually used.

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Pull request closed without merging**